### PR TITLE
[stable/rethinkdb] Support rbac

### PR DESCRIPTION
**Pull request by Malinskiy**
Uses default ServiceAccount by default or creates a new one with permissions to get/list/watch endpoints since the init.sh script requires this to find other rethinkdb instances

### CODE REVIEW

**Comment by Malinskiy** [comment]
CLA is signed, bot needs to recheck

**Comment by unguiculus** [comment]
In the meantime, we have revised our RBAC best practices:
https://github.com/unguiculus/helm/blob/feature/rbac/docs/chart_best_practices/rbac.md

Please update accordingly. Here's an example for a chart that already implements this:
https://github.com/kubernetes/charts/tree/master/stable/cert-manager

File: stable/rethinkdb/templates/_helpers.tpl [hunk 1]
```diff
@@ -14,3 +14,21 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- $name := default .Chart.Name .Values.nameOverride -}}
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "rethinkdb.chart" -}}
```
Comment (unguiculus) [inline stable/rethinkdb/templates/_helpers.tpl:21]: You copied this template but don't use it.

File: incubator/cassandra/templates/cassandra-statefulset.yaml [hunk 2]
```diff
@@ -69,7 +69,7 @@ spec:
         {{- $seed_size := default 1 .Values.config.seed_size | int -}}
         {{- $global := . }}
         - name: CASSANDRA_SEEDS
-          value: "{{- range $i, $e := until $seed_size }}{{ template "cassandra.fullname" $global }}-{{ $i }}.{{ template "cassandra.fullname" $global }}.{{ $global.Release.Namespace }}.svc.cluster.local,{{- end }}"
+          value: "{{- range $i, $e := until $seed_size }}{{ template "cassandra.fullname" $global }}-{{ $i }}.{{ template "cassandra.fullname" $global }}.{{ $global.Release.Namespace }}.{{ $global.Values.clusterDomainSuffix }},{{- end }}"
```
Comment (unguiculus) [inline incubator/cassandra/templates/cassandra-statefulset.yaml]: That's an unrelated change that should not be part of this PR.

Comment (Malinskiy) [reply incubator/cassandra/templates/cassandra-statefulset.yaml]: you're blazing fast. are you a robot? just fixed this :)

**Comment by unguiculus** [comment]
Please add standard labels to RBAC resources.

File: stable/rethinkdb/templates/_helpers.tpl (hunk 1 above)
Comment (Malinskiy) [reply stable/rethinkdb/templates/_helpers.tpl:21]: You're incorrect, I've added this to the ServiceAccount entity, check the label `chart`

Comment (unguiculus) [reply stable/rethinkdb/templates/_helpers.tpl:21]: Could you just use this everywhere for consistency?

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/rethinkdb/templates/_helpers.tpl (hunk 1 above)
Comment (Malinskiy) [reply stable/rethinkdb/templates/_helpers.tpl:21]: yes

Comment (Malinskiy) [reply stable/rethinkdb/templates/_helpers.tpl:21]: done

**Review by unguiculus — APPROVED** [review]

**Comment by unguiculus** [comment]
cc @meenie

**Comment by meenie** [comment]
@unguiculus I haven't read up on RBAC so I'm not sure exactly how it works. If this doesn't break peoples environments if they upgrade and only enhances it, then I suppose it's all good :). I don't want to hold up this PR because of my ignorance on the subject.

Thanks for the ping, I'll make sure do my homework.

**Comment by unguiculus** [comment]
@meenie This shouldn't break anything.
